File: src/Accounting/Vns.Erp.Core.Accounting.Report/Reports/UserControl/KTY_UserControl.vb
Imports Vns.Erp.Core.Admin.Domain

Public Class KTY_UserControl
    Inherits UserControl

    Public Parameter_info As HtBienBaocao = New HtBienBaocao()
    Public Value_info As ValueInfo = New ValueInfo()
    Public NotAllowSendTab As Boolean = False

    Public Overridable Sub ReLoad(ByVal objID As Object)
    End Sub

    Private Sub InitializeComponent()
        Me.SuspendLayout()

        Me.Name = "KTY_UserControl"
        Me.ResumeLayout(False)

    End Sub

    Private Sub KTY_UserControl_KeyDown(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles MyBase.KeyDown
        Try
            Select Case e.KeyCode
                Case Keys.Enter
                    If Not NotAllowSendTab Then
                        SendKeys.Send("{TAB}")
                    End If
                Case Else
            End Select
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub
End Class

File: src/Controls/Vns.Erp.Core.Controls/UserControl/KTY_UserControl.cs
using Vns.Erp.Core.Admin.Domain;
using Vns.Erp.Core;
using System.Windows.Forms;
using System;

namespace Vns.Erp.Core.Controls.UserControl
{
    public class KTY_UserControl : DevExpress.XtraEditors.XtraUserControl
    {
        public HtBienBaocao Parameter_info = new HtBienBaocao();

        public ValueInfo Value_info = new ValueInfo();

        public bool NotAllowSendTab = false;

        public virtual void ReLoad(object objID)
        {
        }

        private void InitializeComponent()
        {
            this.SuspendLayout();
            this.Name = "KTY_UserControl";
            this.ResumeLayout(false);
        }

        private void KTY_UserControl_KeyDown(object sender, System.Windows.Forms.KeyEventArgs e)
        {
            try
            {
                switch (e.KeyCode)
                {
                    case Keys.Enter:
                        if (!NotAllowSendTab) SendKeys.Send("{TAB}");
                        break;
                    default:
                        break;
                }
            }
            catch (Exception ex)
            { }
        }

    }
}